### ISSUE TITLE
math: use types defined in image repository

### ISSUE THREAD

**Issue by crawshaw** [google/gxui#28]
A set of reasonably-common types for image work have ended up in:

https://godoc.org/golang.org/x/image/math/f32

They seem to line up with the types in gxui/math. The only downside is you cannot put methods on types from an external package.

If this seems reasonable, I'll give it a go.

**Comment by shurcooL** [comment]
:+1: to the idea.

If possible, I'd like to see https://godoc.org/github.com/go-gl/mathgl/mgl32 used too. Are there valid reasons why it isn't?

**Comment by ben-clayton** [comment]
@crawshaw is there any reason why the methods couldn't be absorbed into https://godoc.org/golang.org/x/image/math/f32? It's not going to break compatibility.

**Comment by ben-clayton** [comment]
But then again, it's a bit weird to have that sort

**Comment by ben-clayton** [comment]
argh, wrong button

**Comment by ben-clayton** [comment]
But then again, it's a bit weird to have that sort of math in an image library.

**Comment by crawshaw** [comment]
Methods can be added to image/math if we can reach consensus on what they should be. We had a lot of trouble with it in early discussions. We could find common ground on types, so we started there.

**Comment by ben-clayton** [comment]
If the methods cannot be brought across, I don't really see the gain in switching to the f32 package - does it not contain just 7 typedefs? Do you see common functionality between GXUI and f32?

**Comment by crawshaw** [comment]
Let's punt on this issue for a bit, I believe it will be useful but would like to demonstrate it with code.

It is also worth looking at https://godoc.org/github.com/go-gl/mathgl/mgl32 as@shurcooL mentioned, it has a lot of relevant machinery in it.

**Comment by ben-clayton** [comment]
Yup mgl32 looks like a pretty good replacement. It doesn't have my fantastic unicode art though. :)

**Comment by shurcooL** [comment]
> Yup mgl32 looks like a pretty good replacement. It doesn't have my fantastic unicode art though. :)

Please feel free to make a PR, improvements to it are always welcome.
